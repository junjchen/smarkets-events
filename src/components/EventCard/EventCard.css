.event-card {
    border: 1px solid rgb(67, 160, 71);
    margin: 0.5em 0;
    padding: 0.5em;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
}

.event-card:hover {
    color: rgb(255, 255, 255);
    background-color: rgb(67, 160, 71);
}