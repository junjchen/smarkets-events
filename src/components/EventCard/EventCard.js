import React, {Component} from 'react'
import './EventCard.css'

class EventCard extends Component {
    render() {
        const {name, parent_name, id, onSelect} = this.props
        return (
            <section className="event-card" onClick={onSelect.bind(null, id)}>
                <h3 className="event-card__title">{name}</h3>
                <p className="event-card__category">{parent_name}</p>
            </section>
        )
    }
}

export default EventCard