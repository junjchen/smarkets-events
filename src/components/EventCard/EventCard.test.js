import React from 'react'
import {shallow} from 'enzyme'
import EventCard from './EventCard'

const createProps = () => ({
    name: 'name',
    parent_name: 'category',
    onSelect: jest.fn()
})

describe('EventCard', () => {
    it('renders same as snapshot', () => {
        const props = createProps()
        const wrapper = shallow(<EventCard {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })
    it('respond to click event', () => {
        const props = createProps()
        const wrapper = shallow(<EventCard {...props}/>)
        wrapper
            .find('.event-card')
            .simulate('click')
        expect(props.onSelect).toBeCalled()
    })
})
