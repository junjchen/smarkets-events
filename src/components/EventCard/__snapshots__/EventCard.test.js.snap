// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard renders same as snapshot 1`] = `
<section
  className="event-card"
  onClick={[Function]}
>
  <h3
    className="event-card__title"
  >
    name
  </h3>
  <p
    className="event-card__category"
  >
    category
  </p>
</section>
`
